### PR TITLE
Changes to the spam command (Take 2)

### PR DESCRIPTION
**Pull request by Gh0stlyy**
Spam will now add a limit of 50 messages, every 5 seconds to avoid ratelimit, and a cooldown if above 25. If a spam amount is not given, it defaults to 10. Some other things changed as well.

### CODE REVIEW

File: cogs/basic.py [hunk 1]
```diff
@@ -125,13 +125,17 @@ def __init__(self, bot):
 
     @commands.command()
     @commands.guild_only()
+    @command.cooldown(1, 30, BucketType.user)
     async def spam(self,ctx,amount:int=None):
         limit = 50
+        halflimit = limit / 2
         if amount is not None:
             if amount > limit:
                 await ctx.send('Hey! You trying to ratelimit me?! Keep it under {limit}.'.format(limit=limit))
```
Comment (Kitty-Therapist) [inline cogs/basic.py:133]: You can also do

`await ctx.send(f"Hey! You trying to ratelimit me?! Keep it under {limit}."`

File: cogs/basic.py [hunk 2]
```diff
@@ -125,29 +125,41 @@ def __init__(self, bot):
...
+            if amount > limit:
+                await ctx.send('Hey! You trying to ratelimit me?! Keep it under {limit}.'.format(limit=limit))
+        else:
+            while amount > 0:
+                if amount < halflimit:
+                    ctx.command.reset_cooldown(ctx)
+                await ctx.send("You have requested spam.")
+                await asyncio.sleep(5)
+                amount -= 1
+        else:
```
Comment (Kitty-Therapist) [inline cogs/basic.py:139]: The bot is apparently not liking that there are two else's, and so it is spitting out this error here: `D:\Fuzzinator>python bot.py
Traceback (most recent call last):
  File "bot.py", line 112, in <module>
    bot.load_extension(f"cogs.{extension}")
  File "D:\Programming\lib\site-packages\discord\ext\commands\bot.py", line 695, in load_extension
    lib = importlib.import_module(name)
  File "D:\Programming\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 724, in exec_module
  File "<frozen importlib._bootstrap_external>", line 860, in get_code
  File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "D:\Fuzzinator\cogs\basic.py", line 142
    else:
       ^
SyntaxError: invalid syntax
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x04333D10>`

This was after I defined the BucketType which I had been missing on the top. So commenting the beginning of the point.

Comment (Gh0stlyy) [reply cogs/basic.py:139]: Fixed this by moving the first else statement inside
`if amount is not None:`
statement

File: cogs/basic.py [hunk 3]
```diff
@@ -125,29 +125,38 @@ def __init__(self, bot):
...
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
+    @command.cooldown(1, 30, BucketType.user)
+    async def spam(self,ctx,amount:int=None):
```
Comment (AEnterprise) [inline cogs/basic.py:129]: this entire thing is a bit of mess, why not just use an default value of 10? that makes the first if and else redundant and cleans this thing up quite a bit

**Pull request merged**